util/gossiper/torrent_gossiper.py

```diff
@@ -1,28 +1,46 @@
 import os
 import random
+import json
 from torrentool.api import Torrent
 from .gossiper import Gossiper
 from messages import TorrentPayload, MESSAGE_TORRENT_ID
 
 
 class TorrentGossiper(Gossiper):
-    def __init__(self, delay, peers, community, signed=True):
+    def __init__(self, delay, peers, community, signed=True, profiles=[]):
         self.delay = delay
         self.peers = peers
         self.community = community
         self.torrent_list = os.listdir("torrents/")
         self.signed = signed
+        self.profiles = profiles
+
+    def get_magnet_hash(self, magnet: str) -> str:
+        """
+        Extracts the has from a magnetlink
+        """
+        temp = magnet.split("xt=urn:btih:")[1]
+        temp = temp.split("&")[0]
+
+        return temp
 
     # sends random torrents to a set of peers
     def gossip(self):
         for p in self.community.get_peers():
-            torrent_to_send = random.choice(self.torrent_list)
-            torrent = Torrent.from_file(f"torrents/{torrent_to_send}")
+            for profile in self.profiles:
+                data_to_send = json.dumps([
+                    ["Key", self.get_magnet_hash(profile["magnet"])],
```

Comment:
in #62 i updated the profiles to be per video using the hash and video index rather than the magnet in the format `$hash?index=$index`